src/main/java/fi/crosskey/card/vehicle/simulator/service/CarsController.java

```diff
@@ -14,6 +16,7 @@
 @RequestMapping("/cars")
 public class CarsController {
 
+    @Autowired
```

Comment:
Thanks for finding my missing `@Autowire`. In general, prefer autowiring the constructor instead of fields.

So:

```java
@Autowired
public CarsController(final CarRepository carRepository) {
```